tests/regression/56-floats/05-invariant.c

```diff
@@ -0,0 +1,99 @@
+// PARAM: --enable ana.float.interval --enable ana.int.interval
+#include <assert.h>
+#include <float.h>
+
+int main()
+{
+    double a;
+    int b;
+
+    // make a definitly finite!
+    if (b)
+    {
+        a = 100.;
+    }
+    else
+    {
+        a = -100.;
+    };
+
+    if (b != 1)
+    {
+        assert(b != 1); // SUCCESS!
+    }
+    else
+    {
+        assert(b == 1); // SUCCESS!
+    }
+
+    if (a != 1.)
+    {
+        // this would require a exclusion list etc.
+        assert(a != 1.); // UNKNOWN!
+    }
+
+    if (a == 1.)
+    {
+        assert(a == 1.); // SUCCESS!
+    }
+
+    if ((int)a)
+    {
+        assert(a != 0.); // UNKNOWN!
+    }
+    if (a) // here a explicit cast to (int) should be inserted
```

Comment:
I think this second if (a) is superflous here, since the same happens before with an explicit cast

Reply:
But this generates a warning, which can be tracked by having this case. Therefore, i would keep it.

Reply:
ah, ok, I understand